MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/CreateCapillaryPressureModel.cpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file   CreateCapillaryPressureModel.cpp
+ *
+ * Created on November 1, 2016, 10:06 AM
+ */
+
+#include "CreateCapillaryPressureModel.h"
+
+#include <array>
+
+#include "BaseLib/ConfigTree.h"
+#include "BaseLib/Error.h"
+
+#include "CapillaryPressureSaturation.h"
+#include "BrookCoreyCapillaryPressure.h"
+#include "vanGenuchtenCapillaryPressure.h"
+
+namespace MaterialLib
+{
+namespace PorousMedium
+{
+/**
+    \param config ConfigTree object which contains the input data
+                  including <type>BrookCorey</type> or <type>vanGenuchten</type>
+                  and it has a tag of <capillary_pressure>
+*/
+static std::unique_ptr<CapillaryPressureSaturation>
+createBrookCoreyOrVanGenuchten(BaseLib::ConfigTree const& config,
+                               const bool is_brook_corey)
```

Comment:
I'd split it into two different create functions and avoid the boolean parameter.
Adding a third model here would be difficult.